src/python/pants/backend/python/rules/pex_from_targets.py

```diff
@@ -112,18 +112,32 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
         python_setup,
     )
 
-    requirements = PexRequirements.create_from_requirement_fields(
-        (
-            tgt[PythonRequirementsField]
-            for tgt in all_targets
-            if tgt.has_field(PythonRequirementsField)
-        ),
-        additional_requirements=request.additional_requirements,
-    )
+    if python_setup.options.use_all_requirements:
+        if python_setup.requirement_constraints is None:
+            raise ValueError(
+                "--python-setup-use-all-requirements is set, so "
+                "--python-setup-requirement-constraints must be provided, e.g., using the "
```

Comment:
Preferred to use the pants.toml format of `use_all_requirements` in the `[python-setup]` scope.